src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1615,45 +1615,27 @@ const char16_t* Compiler::eeGetShortClassName(CORINFO_CLASS_HANDLE clsHnd)
     return param.classNameWidePtr;
 }
 
-const WCHAR* Compiler::eeGetCPString(size_t strHandle)
+void Compiler::eePrintObjectDescription(const char* prefix, size_t handle)
 {
-#ifdef HOST_UNIX
-    return nullptr;
-#else
-    char buff[512 + sizeof(CORINFO_String)];
-
-    // make this bulletproof, so it works even if we are wrong.
-    if (ReadProcessMemory(GetCurrentProcess(), (void*)strHandle, buff, 4, nullptr) == 0)
-    {
-        return (nullptr);
-    }
-
-    CORINFO_String* asString = nullptr;
-    if (impGetStringClass() == *((CORINFO_CLASS_HANDLE*)strHandle))
-    {
-        // strHandle is a frozen string
-        // We assume strHandle is never an "interior" pointer in a frozen string
-        // (jit is not expected to perform such foldings)
-        asString = (CORINFO_String*)strHandle;
-    }
-    else
+    const size_t maxStrSize = 64;
+    char         str[maxStrSize];
+    size_t       actualLen = this->info.compCompHnd->printObject((void*)handle, str, maxStrSize);
+    if (actualLen == -1)
```

Comment:
`size_t` is unsigned. I expect that you will see some compilers complaining about it on some platforms.

I think it would be better if `print` APIs never fail and always give you back something.

Reply:
Thanks for noticing, fixed. Now it will be just an empty buffer